src/OrchardCore.Modules/OrchardCore.Sitemaps/ViewModels/CustomPathSitemapSourceViewModel.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using OrchardCore.Sitemaps.Models;
+
+namespace OrchardCore.Sitemaps.ViewModels
+{
+    /// <summary>
+    /// A sitemap source for managing custom url.
+    /// </summary>
+    public class CustomPathSitemapSourceViewModel
+    {
+        /// <summary>
+        /// Gets and sets the custom url.
+        /// </summary>
+        [Required]
+        [RegularExpression(@"^~?\/[^?#]+(\?([^#]*))?(#(.*))?")]
```

Comment:
I don't think we should be using RegEx here.

Better to use the same as the `AutorouteDriver` rules.